pkg/tsdb/cloudwatch/time_series_query_test.go

```diff
@@ -136,3 +136,224 @@ func TestTimeSeriesQuery(t *testing.T) {
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "300 Average", resp.Responses["A"].Frames[0].Name)
+	})
+
+	t.Run("where no alias is provided and query is math expression, then frame name is queryId", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricEditorMode: MetricEditorModeRaw,
+		})
+
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "query id", resp.Responses["A"].Frames[0].Name)
+	})
+
+	t.Run("where no alias provided and query type is MetricQueryTypeQuery, then frame name is label", func(t *testing.T) {
+		query := newTestQuery(t, queryParameters{
+			MetricQueryType: MetricQueryTypeQuery,
+		})
+
+		resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+			PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+			Queries: []backend.DataQuery{
+				{
+					RefID:     "A",
+					TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+					JSON:      query,
+				},
+			},
+		})
+
+		assert.NoError(t, err)
+		assert.Equal(t, "response label", resp.Responses["A"].Frames[0].Name)
+	})
+
+	// where query is inferred search expression and not multivalued dimension expression, then frame name is label
+	testCasesReturningLabel := map[string]queryParameters{
+		"with specific dimensions, matchExact false": {Dimensions: queryDimensions{[]string{"some-instance"}}, MatchExact: false},
+		"with wildcard dimensions, matchExact false": {Dimensions: queryDimensions{[]string{"*"}}, MatchExact: false},
+		"with wildcard dimensions, matchExact true":  {Dimensions: queryDimensions{[]string{"*"}}, MatchExact: true},
+		"no dimension, matchExact false":             {Dimensions: queryDimensions{}, MatchExact: false},
+	}
+	for name, parameters := range testCasesReturningLabel {
+		t.Run(name, func(t *testing.T) {
+			query := newTestQuery(t, parameters)
+
+			resp, err := executor.QueryData(context.Background(), &backend.QueryDataRequest{
+				PluginContext: backend.PluginContext{DataSourceInstanceSettings: &backend.DataSourceInstanceSettings{}},
+				Queries: []backend.DataQuery{
+					{
+						RefID:     "A",
+						TimeRange: backend.TimeRange{From: time.Now().Add(time.Hour * -2), To: time.Now().Add(time.Hour * -1)},
+						JSON:      query,
+					},
+				},
+			})
+
+			assert.NoError(t, err)
+			assert.Equal(t, "response label", resp.Responses["A"].Frames[0].Name)
+		})
+	}
+
+	// complementary test cases to above return default of "metricName_stat"
```

Comment:
nice! In future PRs where we use dynamic labels instead, we should probably consider deprecating this and always return `label` in these cases. That would be the expected behavior IMO.